src/main/java/org/opensearch/security/configuration/DlsQueryParser.java

```diff
@@ -15,151 +15,152 @@
 
 package org.opensearch.security.configuration;
 
-import java.io.IOException;
 import java.util.Set;
 import java.util.concurrent.Callable;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.lucene.index.Term;
 import org.apache.lucene.search.BooleanClause.Occur;
 import org.apache.lucene.search.BooleanQuery;
-import org.apache.lucene.search.ConstantScoreQuery;
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.PrefixQuery;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.join.BitSetProducer;
 import org.apache.lucene.search.join.ToChildBlockJoinQuery;
+import org.opensearch.common.xcontent.DeprecationHandler;
 import org.opensearch.common.xcontent.NamedXContentRegistry;
 import org.opensearch.common.xcontent.XContentParser;
 import org.opensearch.common.xcontent.json.JsonXContent;
 import org.opensearch.index.query.AbstractQueryBuilder;
 import org.opensearch.index.query.ParsedQuery;
 import org.opensearch.index.query.QueryBuilder;
 import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.security.queries.QueryBuilderTraverser;
 
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
 
-import static org.opensearch.common.xcontent.DeprecationHandler.THROW_UNSUPPORTED_OPERATION;
 
+public final class DlsQueryParser {
 
-final class DlsQueryParser {
-
+    private static final Logger log = LogManager.getLogger(DlsQueryParser.class);
     private static final Query NON_NESTED_QUERY;
 
     static {
         //Match all documents but not the nested ones
-        //Nested document types start with __
+        //Nested document types start with __ 
```

Comment:
nit-picking: formatting:
```suggestion
        //Nested document types start with __
```

Reply:
Fixed